client/config/config.go

```diff
@@ -97,8 +97,16 @@ type Config struct {
 	StateDir string
 
 	// AllocDir is where we store data for allocations
+	//
+	// In a production environment this should be owned by root with file
+	// mode 0o700.
 	AllocDir string
 
+	// AllocMountsDir is where we bind mount paths from AllocDir for tasks making
+	// use of the unveil file isolation mode. In a production environment this
+	// should be owned  by root with file mode 0o755.
+	AllocMountsDir string
```

Comment:
Should we add website docs for the `alloc_mounts` configuration in this PR?

Reply:
Will do a big PR at the end for all things exec2